docs/concepts/style_guide.md

```diff
@@ -0,0 +1,178 @@
+
+Teardown steps registered outside of tests should use
+[`#step`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#step-instance_method)
+to document and log what they are doing.
+
+## Acceptable Exit Codes
+
+When using the Beaker
+[`on`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Helpers/HostHelpers#on-instance_method)
+method, the default setting is that only an exit code of 0 (zero) will not
+trigger an error. When other exit codes are acceptable, the
+`:acceptable_exit_codes` key with an array of exit codes should be passed in the
+options hash to `#on`. If 0 (zero) is the only acceptable exit code, then the
+`:acceptable_exit_codes` symbol must not be used.
+
+Each example can be considered a bad example for the situation below it:
+```ruby
+# Single 0 exit code allowed
+on(host, "rm #{file_to_rm}")
+# Single non-0 exit code allowed
+on(host, "rm #{file_to_rm}", :acceptable_exit_codes => 1)
+# Multiple exit codes allowed
+on(host, "rm #{file_to_rm}", :acceptable_exit_codes => [0,1])
+# Any exit code allowed
+on(host, "rm #{file_to_rm}", :accept_all_exit_codes => true)
+```
+
+In the last case, when any exit code is allowed, one must follow-up with a valid
+assertion test.
+
+If an exit_code outside of 0 is expected, one must use acceptable_exit_codes so
+the test will fail on the proper assertion and not error at that command.  Allow
+only the minimum expected set of exit codes unless coverage is provided by
+subsequent assertions.
+
+## Outcomes
+
+When to use each, how to format the message:
+
+### `fail_test`, `pass_test`, & `skip_test` 
+
+These can be used anywhere in a test to exit early. A 
+[`skip_test`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Outcomes#fail_test-instance_method)
+between two assertions, for instance, will run the first assertion, raise an
+exception for `skip_test`, run teardown, and then exit as we expect. The same is
+true for
+[fail_test](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Outcomes#fail_test-instance_method).
+
+[pass_test](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Outcomes#fail_test-instance_method)
+should not be used unless fast-passing in a conditional branch.
+
+### Skipping Tests
+
+Skipping tests can be used, for instance, when they are temporarily failing or
+not yet complete.
+
+**Good:**
+```ruby
+skip_test 'should never run on solaris' if agent['platform'] =~ /solaris/
+```
+**Bad:**
+```ruby
+confine :to, :platform => 'solaris:pending'
+```
+
+If they are failing due to an “expected failure” you should wrap your failing
+assertion in an
+[`expect_failure`](http://www.rubydoc.info/github/puppetlabs/beaker/Beaker/DSL/Structure#expect_failure-instance_method)
+block with an explanatory logging message:
+
+```ruby
+expect_failure('expected to fail due to PE-1234') do
+  assert_equal(400, response.code, 'bad response code from API call')
+end
+```
```

Comment:
the above also illustrates why we might need a quick TL;DR on when to (not) use parens in method calls, even though it's spelled out in the ruby community style guide.